Clears some rubocop offenses

harman28 opened the pull request:
Attempts to solve #250

Onumis commented on lib/rubycritic/generators/html/base.rb:
```diff
@@ -45,7 +45,7 @@ def root_directory
 
-        def get_binding
+        def binding
           binding
```
Won't this is lead to an infinite loop?

harman28 replied:
Oops. Fixing.
I didn't perfectly understand the need for this function. Since it's private and can only be called in child classes, wouldn't it be just as effective to use `binding` directly in those classes instead of calling `get_binding`?

harman28 replied:
Leaving this in for now, I don't think I have enough of an understanding to change this.

Onumis replied:
Well, I'd say that we really don't need that method. I'm also not sure.
Give it a try - remove the method and see if the HTML reports are still properly generated.

Onumis commented on lib/rubycritic/source_control_systems/git.rb:
```diff
@@ -21,7 +21,7 @@ def git(arg)
 
       def self.supported?
-        git('branch 2>&1') && $?.success?
+        git('branch 2>&1') && $CHILD_STATUS.success?
```
This is breaking the tests. We must `require "English"` so that this works  (see https://github.com/rubocop-hq/rubocop/issues/1747).

You can add the require to the top level rubycritic module I think.

harman28 replied:
Fixing this.

> You can add the require to the top level rubycritic module
I was going to require only in the files that used that variable. Would requiring in the top module be better for some reason?

Onumis replied:
Just because it might be used elsewhere. But I guess if this is the only file that uses `$CHILD_STATUS` then we can leave the require here.

Onumis replied:
I mean: if `$CHILD_STATUS` is only used on `RubyCritic::SourceControlSystem` then it can be imported required only in this module.

harman28 commented:
@Onumis Updated the branch. Split the changes into multiple commits for each cop fix, and also reverted some that I didn't completely understand.

Eg: `Style/MethodMissing` todo can be fixed with the following:
```
     def self.method_missing(method, *args, &block)
-      configuration.public_send(method, *args, &block)
+      if configuration.respond_to? method
+        configuration.public_send(method, *args, &block)
+      else
+        super
+      end
     end
```
But this causes the reek check to fail, with smell `method_missing manually dispatches method call`. Ideally when overriding method_missing, I believe the practice is to have a short list of methods for which we'd like special behaviour. Current code seems to want to act on all missing methods indiscriminately though.

Onumis commented:
@harman28 as for the `method_missing` issue:
- https://github.com/troessner/reek/blob/master/docs/Manual-Dispatch.md
- https://github.com/rubocop-hq/ruby-style-guide#no-method-missing

I think we should first understand why we need a `method_missing` here.
All the config vars are defined with `attr_accessor` so there are method to read / write each one of them.

Probably we should just refactor how that configuration class works, but that should be a PR on it's own :sweat_smile:
